pupil-spa/src/app/services/submission/submission.service.ts

```diff
@@ -25,7 +25,13 @@ export class SubmissionService {
           }
         },
         (err) => { throw new Error(err); }
-      ).catch((err) => { throw new Error(err); });
+      ).catch((err) => {
```

Comment:
imagine this code being executed by a class of 30 pupils.  It's like a mini Denial Of Service attack.

Please familiarise with the circuit breaker pattern, and implement it here.

You need to take into consideration how long you should wait before trying, and how long you should keep trying for, as they may reach the end of the check before it succeeds.  If they get to the end of the check, then its pointless to continue trying this.

Reply:
I think whats needed here is a service that act as a wrapper around a remote call.  Then it will be trivial to re-use it for other calls: authentication / check submission too.

Reply:
@GuyHarwood @jon-shipley there was an agreement to treat this as a standalone feature for all the api calls within the app.

Reply:
@GeorgeChatzigiannis No.  This code cannot be added as-is.  It's not a separate feature, its a requirement of the check start notification.  Yes, it would be good to re-use elsewhere, but that's just an added benefit of a good implementation.